packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelForm.tsx

```diff
@@ -126,6 +129,20 @@ export default function ColumnModelForm(props: ColumnModelFormProps) {
     [isJsonSubColumn],
   )
 
+  const currentRawMaxSize = (columnModel as ColumnModelFormData).maximumSize
+
+  const currentMaxSize =
+    currentRawMaxSize !== undefined
+      ? parseInt(currentRawMaxSize.toString(), 10)
+      : null
+
+  const recommendedSize = defaultAnnotationModel
+    ? defaultAnnotationModel.maximumSize
+    : null
+
+  const isSmallerThanRecommendedSize =
+    recommendedSize && currentMaxSize ? currentMaxSize < recommendedSize : false
```

Comment:
Can you pull most of this logic out of the component and into a function (or two) like `getRecommendedMaxSize`?